src/main/scala/sparkz/core/network/DeliveryTracker.scala

```diff
@@ -166,12 +176,12 @@ class DeliveryTracker(system: ActorSystem,
   def setReceived(id: ModifierId, sender: ConnectedPeer): Unit =
     tryWithLogging {
       val oldStatus: ModifiersStatus = status(id)
-      require(isCorrectTransition(oldStatus, Invalid), s"Illegal status transition: $oldStatus -> Received")
+      require(isCorrectTransition(oldStatus, Received), s"Illegal status transition: $oldStatus -> Received")
```

Comment:
Sure, I've checked this. This is indeed a typo from when this code was initially added. 
setReceived is invoked when we receive modifier data and pass it to a hodeViewHolder. For all such modifiers, there is a check in NodeViewSynchronizer, to verify that the status == Requested. Otherwise, it is considered spam. Then, in this `setRequested` method, the only valid transition should be Requested -> Recieved